src/models/data.py

```diff
@@ -160,9 +167,13 @@ def calculate_normalizing_array(self, data_vars: List[str]):
         self.normalizing_dict = cast(Dict[str, Dict[str, np.ndarray]], self.normalizing_dict)
 
         mean, std = [], []
+        normalizing_dict_keys = self.normalizing_dict.keys()
         for var in data_vars:
-            mean.append(self.normalizing_dict[var]['mean'])
-            std.append(self.normalizing_dict[var]['std'])
+            for norm_var in normalizing_dict_keys:
+                if var.endswith(norm_var):
```

Comment:
what is this doing?

Reply:
```
                if var.endswith(norm_var):
```

Reply:
Now, we have variables that no longer explicitly match what is in the normalizing dict. For instance, instead of `precip` we have `lat_1_lon_1_precip`. This 'shifted precip' will have the same mean and std, so should be normalized using the same values in the normalizing dict - this achieves that.